### DISCUSSION TITLE
Cannot print the generated HMTL page to PDF with browsers

### DISCUSSION THREAD

**Discussion by BettyJJ** [facelessuser/MarkdownPreview#162] [Q&A] [answered]
Or rather, the printed PDF is blank. I tried Chrome and Edge, with the same results.
Freshly installed this plugin without any customized settings. 
Printing other webpages works fine.
I'm guessing it might be related to the default CSS? (https://stackoverflow.com/a/36761253/802678)

**Comment by facelessuser** [comment, marked as answer]
Maybe? I have no idea exactly what you are trying to print, but there is definitely a possibility that something isn't compatible with printing to PDF. I don't think any time has been spent to ensure compatibility with printing to PDF.

**Reply by BettyJJ** [reply]
After some testing, I'm pretty sure it's the length. If the total length of the printed pdf is within one page, everything is fine. If it exceeds one page, the pdf is blank. The contents (elements) of the Markdown file do not matter.

The whole reason I'm using this plugin is to convert Markdown to PDF. I'm surprised that nobody else uses it for this. 😅 I think it's possible to tweak the CSS a little and make printing to PDF work out of the box and increase the potential user base, but I can also understand if you don't have the time or don't feel the need to support it. Thank you very much anyway!

**Reply by facelessuser** [reply]
It's something to look into. I'd have to first reproduce this issue. I'm having a hard time understanding how length could affect this, and I'm not discounting your experience, but it is something I'd have to look into personally to fully understand.

**Reply by BettyJJ** [reply]
Thank you! Steps to reproduce:

1. install this plugin without any customized settings
2. open any Markdown file that's more than 40 lines long
3. convert to HTML and open the HTML file in Chrome
4. Ctrl+P or right-click and choose "Print"
5. the printed PDF is blank.

**Reply by facelessuser** [reply]
@BettyJJ This may be a bug in Chrome. I was able to reproduce in Chrome, but I could not reproduce in Firefox or Safari. This seems to be Chrome specific, which is weird.

Now, I'm not saying there isn't something we can fix here, but so far, I am uncertain what the issue is, especially since this only happens in Chrome.

The content is still there, you can see here I've selected the text in the preview. It's there, but weirdly invisible.

<img width="1119" alt="Screenshot 2023-02-13 at 1 33 25 PM" src="https://user-images.githubusercontent.com/1055125/218568709-c9a09211-13d3-458a-8238-99384cd9fc10.png">

**Reply by BettyJJ** [reply]
This also happens in Microsoft Edge. I did not test on Firefox or Safari.
Yes, the content is still there. If I remove the styles, the PDF can be printed normally. So this must be something related to the default CSS.

**Reply by facelessuser** [reply]
I think I figured it out. I have no idea why this breaks things, but I have a fix.

At one time, we drew an outline around the content with a black border. At some point, it somehow got changed to white. Why? I have no idea. The intent was to remove the border, but maybe I was playing around to see what it looked like removed by setting it to white.

Anyways, it got committed, and for some reason, that makes Chrome turn all the text white in the preview, but only if the content spills out to two pages.

The fix, just remove the border as was originally intended.

I'll have a fix in the next release. If you must do previews before then, use Firefox or (if on a mac) Safari.

**Reply by facelessuser** [reply]
By the way, Edge is chrome based, so yeah, still chrome 😄

**Reply by BettyJJ** [reply]
Thanks! So happy we found the fix! 😄

**Reply by facelessuser** [reply]
Didn't think I was going to find it 🙃

**Reply by facelessuser** [reply]
Release made

**Reply by facelessuser** [reply]
Change may take an hour or two to be available on PackageControl
